modules/core/src/main/scala/ch/epfl/scala/debugadapter/internal/stepfilter/ScalaStepFilter.scala

```diff
@@ -14,8 +14,25 @@ import ch.epfl.scala.debugadapter.ScalaVersion
 import java.util.Optional
 abstract class ScalaStepFilter(scalaVersion: ScalaVersion) extends StepFilter {
   protected def skipScalaMethod(method: Method): Boolean
-  override def formatName(method: Method): Optional[String] = {
-    Optional.of(method.name())
+  def format(method: Method): Optional[String] = {
```

Comment:
You should try to use `Option` where possible. `Optional` is only needed to make the bridge between Scala and Java or between two Scala versions.

```suggestion
  def format(method: Method): Option[String] = {
```